Tests/PlayerTests/Mocks/PlaybackEngine/Internal/Events/AsyncschedulerSpy.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+@testable import Player
+
+class AsyncSchedulerSpy: AsyncScheduler {
+	var code: (() async -> Void)?
+	
+	init(code: @escaping () async -> Void) {
+		self.code = code
+	}
+	
+	static func schedule(code: @escaping () async -> Void) -> AsyncScheduler {
+		return AsyncSchedulerSpy(code: code)
+	}
+	
+	func execute() async {
+		await code?()
+	}
+}
+
+class AsyncSchedulerFactorySpy: AsyncSchedulerFactory {
```

Comment:
[suggestion] it can be `final`, right?
```suggestion
final class AsyncSchedulerFactorySpy: AsyncSchedulerFactory {
```
[suggestion 2] we could also split them in different files for better discoverability.